src/Campaigns/Repositories/CampaignRepository.php

```diff
@@ -76,10 +76,13 @@ public function insert(Campaign $campaign): void
 
         Hooks::doAction('givewp_campaign_creating', $campaign);
 
-        $dateCreated = Temporal::withoutMicroseconds($campaign->createdAt ?: Temporal::getCurrentDateTime());
+        $currentDate = Temporal::getCurrentDateTime();
+
+        $dateCreated = Temporal::withoutMicroseconds($campaign->createdAt ?: $currentDate);
         $dateCreatedFormatted = Temporal::getFormattedDateTime($dateCreated);
-        $startDateFormatted = Temporal::getFormattedDateTime($campaign->startDate);
-        $endDateFormatted = Temporal::getFormattedDateTime($campaign->endDate);
+
+        $startDateFormatted = Temporal::getFormattedDateTime($campaign->startDate ?: $currentDate);
+        $endDateFormatted = $campaign->endDate ? Temporal::getFormattedDateTime($campaign->endDate) : $campaign->endDate;
```

Comment:
Same here 

```suggestion
        $endDateFormatted = $campaign->endDate ? Temporal::getFormattedDateTime($campaign->endDate) : null;
```

Reply:
Resolved in 43c8a514386deb2e26b564708d89e2bfc1464b82